Mathlib/Analysis/Analytic/Basic.lean

```diff
@@ -366,8 +364,62 @@ theorem radius_le_radius_continuousLinearMap_comp (p : FormalMultilinearSeries 
   simpa only [norm_norm] using f.norm_compContinuousMultilinearMap_le (p n)
 #align formal_multilinear_series.radius_le_radius_continuous_linear_map_comp FormalMultilinearSeries.radius_le_radius_continuousLinearMap_comp
 
+/-- The radius of the Cartesian product of two formal series is the minimum of their radii. --/
+lemma radius_prod_eq_min
+    (p : FormalMultilinearSeries 𝕜 E F) (q : FormalMultilinearSeries 𝕜 E G) :
+    (p.prod q).radius = min p.radius q.radius := by
+  apply le_antisymm
+  · refine ENNReal.le_of_forall_nnreal_lt fun r hr => ?_
+    rw [le_min_iff]
+    have := (p.prod q).isLittleO_one_of_lt_radius hr
+    constructor
+    all_goals { -- kludge, there is no "work_on_goal" in Lean 4?
+      apply FormalMultilinearSeries.le_radius_of_isBigO
+      refine (isBigO_of_le _ fun n ↦ ?_).trans this.isBigO
+      rw [norm_mul, norm_norm, norm_mul, norm_norm]
+      refine mul_le_mul_of_nonneg_right ?_ (norm_nonneg _)
+      rw [FormalMultilinearSeries.prod, ContinuousMultilinearMap.op_norm_prod]
+      try apply le_max_left
+      try apply le_max_right }
+  · refine ENNReal.le_of_forall_nnreal_lt fun r hr => ?_
+    rw [lt_min_iff] at hr
+    have := ((p.isLittleO_one_of_lt_radius hr.1).add
+      (q.isLittleO_one_of_lt_radius hr.2)).isBigO
+    refine (p.prod q).le_radius_of_isBigO ((isBigO_of_le _ λ n ↦ ?_).trans this)
+    rw [norm_mul, norm_norm, ←add_mul, norm_mul]
+    refine mul_le_mul_of_nonneg_right ?_ (norm_nonneg _)
+    rw [FormalMultilinearSeries.prod, ContinuousMultilinearMap.op_norm_prod]
+    refine (max_le_add_of_nonneg (norm_nonneg _) (norm_nonneg _)).trans ?_
+    apply Real.le_norm_self
+
```

Comment:
This proof and the next are a bit convoluted, but I couldn't find a nicer way of doing it.